internal/infrastructure/mongo/config.go

```diff
@@ -3,44 +3,114 @@ package mongo
 import (
 	"context"
 	"errors"
+	"math/rand"
+	"time"
 
+	"github.com/google/uuid"
 	"github.com/reearth/reearth-backend/internal/infrastructure/mongo/mongodoc"
 	"github.com/reearth/reearth-backend/internal/usecase/repo"
 	"github.com/reearth/reearth-backend/pkg/config"
+	"github.com/reearth/reearth-backend/pkg/log"
 	"github.com/reearth/reearth-backend/pkg/rerror"
 	"go.mongodb.org/mongo-driver/bson"
 	"go.mongodb.org/mongo-driver/mongo"
 	"go.mongodb.org/mongo-driver/mongo/options"
 )
 
-var upsert = true
+var upsert = false
 
 type configRepo struct {
 	client *mongodoc.ClientCollection
+	lockID *uuid.UUID
 }
 
+type Config struct {
```

Comment:
```suggestion
type ConfigDoc struct {
```

It may confuse with `pkg/config.Config`